apps/macro-sandbox/lib/wrappers/wrapper.py

```
@@ -0,0 +1,323 @@
+    compiled_code = compile(wrapped_code, script_path, 'exec')
+except Exception as e:
+    print(json.dumps({"status": "error", "results": [], "errors": [f"Failed to compile script: {str(e)}"]}))
+    sys.exit(0)
+
+class SafeModule:
+    """Blocks introspection on wrapped modules."""
+    BLOCKED_ATTRS = {
+        '__globals__', '__code__', '__builtins__', '__dict__', 
+        '__class__', '__bases__', '__subclasses__', '__init__',
+        '__loader__', '__spec__', '__package__'
+    }
```

Ruff (0.15.9)

[warning] Mutable default value for class attribute

(RUF012)